redis/redis.go

```diff
@@ -188,14 +226,28 @@ func (store *RedisFeatureStore) Upsert(key string, f ld.Feature) error {
 
 	_, err := c.Do("HSET", store.featuresKey(), key, data)
 
+	if err == nil && store.cache != nil {
+		store.cache.Set(key, f, store.timeout)
+	}
+
 	return err
 }
 
 func (store *RedisFeatureStore) Initialized() bool {
+	if store.cache != nil {
+		if _, present := store.cache.Get("$initialized$"); present {
+			return true
+		}
+	}
+
 	c := store.getConn()
 	defer c.Close()
 
 	init, err := r.Bool(c.Do("EXISTS", store.featuresKey()))
 
+	if store != nil && err == nil && init {
```

Comment:
I decided to only make this a positive cache. If the store hasn't been initialized, this will hit redis more aggressively.